EIPS/eip-3005.md

```diff
@@ -0,0 +1,340 @@
+---
+eip: 3005
+title: Batched meta transactions
+author: Matt (@defifuture)
+discussions-to: https://ethereum-magicians.org/t/eip-3005-the-economic-viability-of-batched-meta-transactions/4673 
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-09-25
+---
+
+## Simple Summary
+
+Defines an extension function for ERC-20 (and other fungible token standards), which allows receiving and processing a batch of meta transactions.
+
+![](../assets/eip-3005/meta-txs-directly-to-token-smart-contract.png)
+
+## Abstract
+
+This EIP proposes a new function called `processMetaBatch()` that extends any fungible token standard, and enables batched meta transactions coming from many senders in one on-chain transaction. 
+
+The function must be able to receive multiple meta transactions data and process it. This means validating the data and the signature, before proceeding with token transfers based on the data.
+
+The function enables senders to make gasless transactions, while reducing the relayer's gas cost due to batching.
+
+## Motivation
+
+Meta transactions have proven useful as a solution for Ethereum accounts that don't have any ether, but hold ERC-20 tokens and would like to transfer them (gasless transactions).
+
+The current meta transaction relayer implementations only allow relaying one meta transaction at a time. Some also allow batched meta transactions from the same sender. But none offers batched meta transactions from **multiple** senders.
+
+The motivation behind this EIP is to find a way to allow relaying batched meta transactions from **many senders** in **one on-chain transaction**, which also **reduces the total gas cost** that a relayer needs to cover.
+
+## Specification
+
+### Meta transaction data
+
+In order to successfully validate and transfer tokens, the proposed function needs to process the following data about a meta transaction:
```

Comment:
> the proposed function

EIPs don't propose, they assert. Write this assuming the person reading it has already decided to implement this and they are just looking to see what to implement.  Think email addresses, when you read the RFC for them to correctly implement an email address parser you are expecting to see assertions rather than proposals.